util/src/main/scala/io/guanaco/alerta/util/AlertaRouteBuilderSupport.scala

```diff
@@ -29,9 +30,19 @@ trait AlertaRouteBuilderSupport { self: RouteBuilder =>
 
       definition
         .process(new Processor {
-          override def process(exchange: Exchange): Unit =
+          override def process(exchange: Exchange): Unit = {
             if (!exchange.getUnitOfWork.containsSynchronization(synchronization))
               exchange.getUnitOfWork.addSynchronization(synchronization)
+
+            val body = exchange.getIn.getBody
+            try {
+              if (!exchange.getProperties.containsKey(RESOURCE_PROPERTY)) {
+                Option(body) map { value => exchange.setProperty(RESOURCE_PROPERTY, config.resource(value.asInstanceOf[T])) }
+              }
+            } catch {
+              case e: ClassCastException => exchange.setProperty(RESOURCE_PROPERTY, s"UnmappedType:${body.getClass.getSimpleName}")
+            }
+          }
```

Comment:
Dit is bijna letterlijk dezelfde code als hetgeen er in de top-level `configureAlerta` method hieronder. Misschien beter een  inner Processor class van maken die je kan herbruiken op die twee plaatsen.